nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/TailFile.java

```diff
@@ -610,7 +625,13 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         }
 
         for (String tailFile : states.keySet()) {
-            processTailFile(context, session, tailFile);
+            try {
+                processTailFile(context, session, tailFile);
+            } catch (NulCharacterEncounteredException e) {
+                getLogger().warn("NUL character encountered in " + tailFile + " and '" + REREAD_ON_NUL.getDisplayName() + "' is set to 'true', yielding.");
+                context.yield();
+                return;
```

Comment:
It is intentional. Your suggestion makes sense and would be a bit better but I think the complexity it would require is not worth it. This should be a rare scenario.

Reply:
As it might be unexpected for the users, I would mention this in the documentation. Also, if you are not willing to go with it, please create a ticket as a possible further improvement. (It might or might not be covered but I think would be good to not forgot about it)

Reply:
I'll update the description of the property to note this behaviour.
If the behaviour itself is deemed to be fine, I'd rather not add a ticket (that would probably wouldn't be taken care of anytime soon).